kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/model/VirtualClusterModel.java

```diff
@@ -260,7 +260,7 @@ private static void validatePortUsage(ClusterNetworkAddressConfigProvider cluste
     }
 
     private static void validateTLsSettings(ClusterNetworkAddressConfigProvider clusterNetworkAddressConfigProvider, Optional<Tls> tls) {
-        if (clusterNetworkAddressConfigProvider.requiresTls() && (tls.isEmpty() || !tls.get().definesKey())) {
+        if (clusterNetworkAddressConfigProvider.requiresServerNameIndication() && (tls.isEmpty() || !tls.get().definesKey())) {
```

Comment:
Sonar is pointing out that this isn't very well covered, we have a `missingTls` test in `KafkaProxyTest` which covers the `tls.isEmpty()`, maybe add another for `tls` with no key.

Reply:
Thanks, I'll fix it tomorrow